packages/reco/interface/SRecEvent.cxx

```diff
@@ -590,16 +590,33 @@ void SRecTrack::print(std::ostream& os) const
   os << "Chi square at vertex: " << fChisqVertex << std::endl;
 }
 
-void SRecDimuon::calcVariables()
+void SRecDimuon::calcVariables(int choice)
 {
+    TLorentzVector pos, neg;
+    if(choice == 0)
+    {
+        pos = p_pos;
+        neg = p_neg;
+    }
+    else if(choice == 1)
+    {
+        pos = p_pos_target;
+        neg = p_neg_target;
+    }
+    else if(choice == 2)
+    {
+        pos = p_pos_dump;
+        neg = p_neg_dump;
+    }
```

Comment:
Can you add an "else" case to show a warning (or an error to abort) if `choice` is invalid?

Reply:
Good advice, added another commit